core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -798,6 +803,12 @@ class Partition(val topicPartition: TopicPartition,
       // to maintain the decision maker controller's epoch in the zookeeper path
       controllerEpoch = partitionState.controllerEpoch
 
+      val isNewLeaderEpoch = partitionState.leaderEpoch > leaderEpoch
```

Comment:
Can we add a bit more detail in the description about why we moved this to be before updateAssignmentAndIsr.